utils/config_stores.py

```diff
@@ -0,0 +1,237 @@
+# std
+import itertools
+import logging
+import simplejson as json
+from os import path
+
+# 3p
+from consul import Consul
+from etcd import EtcdKeyNotFound
+from etcd import Client as etcd_client
+from urllib3.exceptions import TimeoutError
+
+
+log = logging.getLogger(__name__)
+
+DEFAULT_ETCD_HOST = '127.0.0.1'
+DEFAULT_ETCD_PORT = 4001
+DEFAULT_ETCD_PROTOCOL = 'http'
+DEFAULT_RECO = True
+DEFAULT_TIMEOUT = 5
+SD_TEMPLATE_DIR = '/datadog/check_configs'
+
+DEFAULT_CONSUL_HOST = '127.0.0.1'
+DEFAULT_CONSUL_PORT = 8500
+DEFAULT_CONSUL_TOKEN = None
+DEFAULT_CONSUL_SCHEME = 'http'
+DEFAULT_CONSUL_CONSISTENCY = 'default'
+DEFAULT_CONSUL_DATACENTER = None
+DEFAULT_CONSUL_VERIFY = True
+
+AUTO_CONF_IMAGES = {
+    # image_name: [check_name, class_name]
+    'redis': ['redisdb', 'Redis'],
```

Comment:
Nitpick but instead of hardcoding the Class name you could just look at the first class that inherits from `AgentCheck` (similarly to what's done in `config.py`)